src/app/pages/system/network/components/default-gateway-dialog/default-gateway-dialog.component.ts

```diff
@@ -76,9 +106,53 @@ export class DefaultGatewayDialog {
     private validatorsService: IxValidatorsService,
   ) {}
 
+  ngOnInit(): void {
+    this.currentGateway$.pipe(untilDestroyed(this)).subscribe((state) => {
```

Comment:
Since you are loading values for the form here, it's better to get rid of loading state approach in the template.
1. Extract this code to new method.
2. Just make normal call to `this.api.call` with error handling and loader like in other forms.
3. Collapse statements that patch form values.
4. Change `currentGateway`, `currentDns1` and `currentDns2` that are currently unused to signals and use them in the template instead of `ixWithLoadingState`